agent_build_refactored/managed_packages/files/bin/python3

```diff
@@ -26,8 +26,11 @@
 
 set -e
 
-DEPENDENCIES_ROOT="/opt/scalyr-agent-2-dependencies"
-DEPENDENCIES_LIBS="${DEPENDENCIES_ROOT}/lib"
+PACKAGE_OPT_DIR="/opt/scalyr-agent-2"
+PACKAGE_LIBS_DIR="${PACKAGE_OPT_DIR}/lib"
+PYTHON_DIR="${PACKAGE_OPT_DIR}/python3"
+PYTHON_LIB_DIR="${PYTHON_DIR}/lib"
 
-export LD_LIBRARY_PATH="${DEPENDENCIES_LIBS}:${DEPENDENCIES_LIBS}/openssl/current:${LD_LIBRARY_PATH}"
-exec "${DEPENDENCIES_ROOT}/bin/python3-original" "$@"
+
+export LD_LIBRARY_PATH="${PYTHON_LIB_DIR}:${PACKAGE_LIBS_DIR}/openssl/current:${LD_LIBRARY_PATH}"
```

Comment:
Why is this needed?  I understand it's using a bundled version of openssl but couldn't the Scalyr agent package have openssl as a dependency to avoid this?

Reply:
> but couldn't the Scalyr agent package have openssl as a dependency to avoid this?

I think it just won'y help us so much. If we specify the dependency just  like `libssl` without any versioning, then we still have to verify if the version provided by the package manager is appropriate so we can fallback to the embedded
openssl if not.

If we specify the dependency `libssl` with version, aka `libssl=>1.1.1`, then we'll just leave behind everyone who can not satisfy that requirement, because now package manager won't simple allow to install this package, so they don't even have a chance to fall back to the embedded openssl.